saleor/plugins/webhook/plugin.py

```diff
@@ -2918,17 +2936,53 @@ def payment_gateway_initialize_session(
             gateways = {gateway.app_identifier: gateway for gateway in payment_gateways}
             apps_identifiers = list(gateways.keys())
 
-        webhooks = get_webhooks_for_event(
-            WebhookEventSyncType.PAYMENT_GATEWAY_INITIALIZE_SESSION,
-            apps_identifier=apps_identifiers,
+        webhooks = get_webhooks_for_event(event_type, apps_identifier=apps_identifiers)
+        request = initialize_request(
+            self.requestor, sync_event=True, event_type=event_type
         )
 
-        request = initialize_request(
-            self.requestor,
-            sync_event=True,
-            event_type=WebhookEventSyncType.PAYMENT_GATEWAY_INITIALIZE_SESSION,
+        pregenerated_subscription_payloads: dict[int, dict[str, dict[str, Any]]] = (
+            defaultdict(lambda: defaultdict(dict))
         )
 
+        app_ids = {webhook.app_id for webhook in webhooks}
+        apps_map = App.objects.filter(id__in=app_ids).in_bulk()
```

Comment:
`get_webhooks_for_event` adds `.select_related("app")` so making `webhook.app` would give you a cached result. So no need to make an additional query here, as you already have this data.

Reply:
I think you can drop these lines, and in 2958 just make `app = webhook.app` 🤔

Reply:
True, it's a leftover from a dataloader.